lasagne/layers/base.py

```diff
@@ -23,8 +23,10 @@ class Layer(object):
     network's output :class:`Layer` instance doubles as a handle to the full
     network.
     """
-    def __init__(self, input_layer):
-        self.input_layer = input_layer
+    def __init__(self, input_layer, name=None):
+        if input_layer is not None:
+            self.input_layer = input_layer
```

Comment:
Ah, I see. It's because `InputLayer` didn't have to call the super constructor before. But then we'll have the same problem with `MultipleInputsLayer`, which shouldn't have a `self.input_layer` member either. Maybe `InputLayer` and `MultipleInputsLayer` shouldn't call the super constructor, but take care of `name` by themselves? Or we need a common base class for `Layer`, `MultipleInputsLayer` and `InputLayer` that takes care of this?

The current solution is a bit dangerous because it means whenever I pass `None` as the first argument of a `DropoutLayer`, for example, it will end up without an `input_layer` property; instead this should be caught and lead to an exception.